tests/test_core.py

```diff
@@ -0,0 +1,30 @@
+import datetime
+import pytest
+
+
+@pytest.mark.parametrize(
+    "table",
+    [
+        ("core__condition"),
+        ("core__documentreference"),
+        ("core__encounter"),
+        ("core__observation"),
+        ("core__patient"),
+        ("core__count_condition_month"),
+        ("core__count_documentreference_month"),
+        ("core__count_encounter_month"),
+        ("core__count_observation_lab_month"),
+        ("core__count_patient"),
+    ],
+)
+def test_core_tables(mock_db_core, tmp_path, table):
+    cursor = mock_db_core.cursor()
+    table_rows = cursor.execute(f"SELECT * FROM {table}").fetchall()
+    # For regenerating data if needed
+    # with open(f'./tests/test_data/core/{table}.txt','wt', encoding="UTF-8") as f:
+    #    f.write(str(table_rows))
+    with open(f"./tests/test_data/core/{table}.txt", "r", encoding="UTF-8") as f:
+        ref_table = eval(f.read())
```

Comment:
nit: What about comparing the csv output? That loses some metadata like typing... But would test the export and is a more familiar human-readable-file-format. This is fine though.

Reply:
lemme sleep on it - my first instinct was no, because:
- this is testing more tables than the export - it might be easier to narrow down an issue by also running the base tables
- Export as a mechanism is already covered in the cli interface test

i :could: see having a test that is like 'get every table that has `__count_` in it and verify they are all in the export list of the manifest?

I went back and forth on making things more readable, and I could go back to trying that - mostly the issue is datetime serialization - but i could write out/read in one row at a time?

Reply:
Hah nice :smile: